skyplane/compute/gcp/gcp_key_manager.py

```diff
@@ -0,0 +1,56 @@
+import subprocess
+from pathlib import Path
+
+from skyplane import exceptions as skyplane_exceptions
+from skyplane import key_root
+from skyplane.utils import logger
+
+
+class GCPKeyManager:
+    """Stores SSH keys for access to GCP VMs."""
+
+    def __init__(self, local_key_dir: Path = key_root / "gcp"):
+        self.local_key_dir = local_key_dir
+
+    def get_private_key(self, key_name: str) -> Path:
+        """Returns path to local keyfile."""
+        return self.local_key_dir / f"{key_name}.pem"
+
+    def get_public_key(self, key_name: str) -> Path:
+        """Returns path to local public keyfile."""
+        return self.local_key_dir / f"{key_name}.pub"
+
+    def key_exists_local(self, key_name: str) -> bool:
+        """Checks if a key exists locally."""
+        private_key_exists = self.get_private_key(key_name).exists() and self.get_private_key(key_name).is_file()
+        public_key_exists = self.get_public_key(key_name).exists() and self.get_public_key(key_name).is_file()
+        return private_key_exists and public_key_exists
+
+    def make_key_local(self, key_name: str) -> Path:
+        """Creates a key locally. GCP does not require registering keys other than in instance metadata."""
+        if self.key_exists_local(key_name):
+            logger.error(f"Key {key_name} already exists locally")
+            raise skyplane_exceptions.PermissionsException(
+                f"Key {key_name} already exists locally, please delete it first or use a different key name."
+            )
+        local_key_file_pem, local_key_file_pub = self.get_private_key(key_name), self.get_public_key(key_name)
+        logger.fs.debug(f"[GCP] Creating local keypair {key_name}")
+        self.local_key_dir.mkdir(parents=True, exist_ok=True)
+        subprocess.Popen(["ssh-keygen", "-t", "rsa", "-b", "4096", "-f", f"{local_key_file_pem}", "-P", "", "-C", f"{key_name}"]).wait()
```

Comment:
Unless we include `ssh-keygen` inside of some dockerized setup, we cannot guarantee its portability. I would recommend using a library like [`cyrptography`](https://cryptography.io/en/latest/) for generating `rsa` keys without calling `ssh-keygen`.

Additionally, we might want to use [`ed25519`](https://cryptography.io/en/latest/hazmat/primitives/asymmetric/ed25519/)

Reply:
Agreed that we should use vendor neutral logic to generate keys. However, ed25519 has issues with Azure which seems to only support RSA keys.

Reply:
Done, added https://github.com/skyplane-project/skyplane/blob/1eee0491852ef4b46159c0a1667f73acdbd0196e/skyplane/compute/key_utils.py